src/3rd_party_add.c

```diff
@@ -174,7 +174,7 @@ enum swupd_code third_party_add_main(int argc, char **argv)
 	globals.no_scripts = true;
 	struct list *bundle_to_install = NULL;
 	bundle_to_install = list_append_data(bundle_to_install, "os-core");
-	ret_code = bundle_add(bundle_to_install, repo_version);
+	ret_code = bundle_add(bundle_to_install, repo_version, NULL, NULL);
```

Comment:
Multiple NULL or boolean parameters are hard to read in a function. In cases like that it's preferable to have 2 functions with different number of arguments (sad that C doesn't support that with the same name).
So I would create
```

 enum swupd_code bundle_add_full(struct list *bundles_list, int version, extra_proc_fn_t pre_add_fn, extra_proc_fn_t post_add_fn); //or bundle_add_extra
 enum swupd_code bundle_add(struct list *bundles_list, int version)
{
    return bundle_add_full(bundles_list, version, NULL, NULL);
}
```